### DISCUSSION TITLE
Output value beetween 0 and 100 % possible?

### DISCUSSION THREAD

**Discussion by ThomasCr** [ScratMan/HASmartThermostat#275] [General]
Hi,

I dont understand what I have made wrong.... but I get only output values with 0 and 100%.
My config looks like:

```
  - platform: smart_thermostat
    name: Thermostat Wohnzimmer Oben
    unique_id: thermostat_wohnzimmer_oben
    heater:
      - number.stellwert_wohnzimmer_oben
    target_sensor: sensor.wohnzimmer_oben_temperature
    outdoor_sensor: sensor.gw1100a_v2_2_5_outdoor_temperature
    #min_temp: 7 # (default: 7)
    max_temp: 27 # (default: 35).
    ac_mode: False
    target_temp: 19
    keep_alive:
      seconds: 60
    # sampling_period (Optional): interval between two computation of the PID. If set to 0, PID computation is called each time the temperature sensor sends an update. Can be float in seconds or time hh:mm:ss (default 0).
    away_temp: 15
    boost_temp: 22.5
    preset_sync_mode: "sync"
    kp: 4 # Set PID parameter, proportional (p) control value (float, default 100).
          # Note: Once the thermostat has been created, changing the configuration PID values will not update 
          # the PID controller. Use the smart_thermostat.set_pid_gain service to update the PID values instead.
    ki: 0.01 # Set PID parameter, integral (i) control value (float, default 0).
    kd: 0 # Set PID parameter, derivative (d) control value (float, default 0).
    ke: 0 # Set outdoor temperature compensation gain (e) control value (float, default 0).
    # ke (Optional): Set outdoor temperature compensation gain (e) control value (float, default 0).
    # force_off_state (Optional): If set to true (default value), Home Assistant will force the heater entity to OFF state when the thermostat is in OFF. Set parameter to false to control the heater entity externally while the thermostat is OFF.
    autotune: "ziegler-nichols"
    #output_min: 0 # default 0
    #output_max: 99 # default 100
    sampling_period: 00:05:00 # By default, the PID controller will be called each time the target sensor is updated. When using main powered sensor with high sampling rate, the sampling_period parameter should be used to slow down the PID controller refresh rate.
    pwm: 0 # Set period of the pulse width modulation. If too long, the response time of the thermostat will be too slow,
          # leading to lower accuracy of temperature control. Can be float in seconds or time hh:mm:ss (default 15mn).
          # Set to 0 when using heater entity with direct input of 0/100% values like valves or lights.
```

Have I understood this manual right, that it is only possible to have 0 OR 100%??

> Set PID mode: smart_thermostat.set_pid_mode
> Use this service to set the PID mode to either 'auto' or 'off'.
> When in auto, the PID will modulate the heating based on temperature value and variation. **When in off, the PID output will be 0% if temperature is above the set point, and 100% if temperature is below the set point.**

![grafik](https://github.com/user-attachments/assets/b6c17b18-d5ab-4420-8726-4cafa0af5061)

It should look more than this with setpoint to 20.5 °C (this is controlled by a PI implementation)

![grafik](https://github.com/user-attachments/assets/f8e54d6c-44d4-4b04-b661-af0db1ecc7c4)

**Comment by ScratMan** [comment]
Hello, you enabled the autotune, please comment this parameter and restart HA

**Reply by ThomasCr** [reply]
Have I also tested now., it makes the same thing - Jump from 0 to 100, waiting a minute and then again 0, and repeating, with different waiting time, Maybee the wrong params for Ki/Kd/.... thats why I wanted to make the autotune....

**Reply by ThomasCr** [reply]
Any other Idea?

**Reply by ThomasCr** [reply]
I updatet my question, with some screenshots. There we can see, that whatever I try, I only get 0 OR 100 as output and I don't understand why.
I started to set it like this - that's the values from my old PI controller (not HA):
```
kp: 4
ki: 60
kd: 0
ke: 0
```
But no - don't work... also with only `kp: 4` and others to zero.

For ref, my last values from my climate entity:
```
hvac_modes:
  - heat
  - "off"
min_temp: 7
max_temp: 27
preset_modes:
  - none
  - away
  - boost
current_temperature: 20.9
temperature: 21.5
hvac_action: idle
preset_mode: none
away_temp: 15
eco_temp: null
boost_temp: 22.5
comfort_temp: null
home_temp: null
sleep_temp: null
activity_temp: null
control_output: 100
kp: 4
ki: 60
kd: 0
ke: 0
pid_mode: "off"
pid_i: 0
friendly_name: Thermostat Kinderzimmer
supported_features: 401
```

**Reply by ScratMan** [reply]
The pid mode is off, the auto-tune disabled it. Please run the smart_thermostat.set_pid_mode action in developer options "Actions" tab, and set the mode to "auto" for your climate entity.

**Reply by ScratMan** [reply]
![Screenshot_20241230-080311](https://github.com/user-attachments/assets/c5832677-c435-4f41-aa16-78685bc60a8f)

**Reply by ThomasCr** [reply]
have I done - but i was thinking, that we need to set pwm to auto, when the HASmartThermostat should imitate the PWM with on and off... thats how I have understand the documentation...

**Reply by ScratMan** [reply]
No, PID mode Off makes the thermostat operate like a standard thermostat without PID.
